gazebo_ros2_control/src/gazebo_ros2_control_plugin.cpp

```diff
@@ -161,44 +165,49 @@ void GazeboRosControlPlugin::Load(gazebo::physics::ModelPtr parent, sdf::Element
   }
 
   // Get robot_description ROS param name
-  if (impl_->sdf_->HasElement("robot_param")) {
-    impl_->robot_description_ = impl_->sdf_->GetElement("robot_param")->Get<std::string>();
+  if (sdf->HasElement("robot_param")) {
+    impl_->robot_description_ = sdf->GetElement("robot_param")->Get<std::string>();
   } else {
     impl_->robot_description_ = "robot_description";  // default
   }
 
   // Get robot_description ROS param name
-  if (impl_->sdf_->HasElement("robot_param_node")) {
+  if (sdf->HasElement("robot_param_node")) {
     impl_->robot_description_node_ =
-      impl_->sdf_->GetElement("robot_param_node")->Get<std::string>();
+      sdf->GetElement("robot_param_node")->Get<std::string>();
   } else {
     impl_->robot_description_node_ = "robot_state_publisher";  // default
   }
 
   // Get the robot simulation interface type
-  if (impl_->sdf_->HasElement("robot_sim_type")) {
-    impl_->robot_hw_sim_type_str_ = impl_->sdf_->Get<std::string>("robot_sim_type");
+  if (sdf->HasElement("robot_sim_type")) {
+    impl_->robot_hw_sim_type_str_ = sdf->Get<std::string>("robot_sim_type");
   } else {
-    impl_->robot_hw_sim_type_str_ = "gazebo_ros2_control/DefaultRobotHWSim";
+    impl_->robot_hw_sim_type_str_ = "gazebo_ros2_control/GazeboSystem";
```

Comment:
I do understand the logic within the `else` branch, that we are looking for a `gazebo_ros2_control/GazeboSystem` interface. But what does the `robot_sim_type` indicate?

Reply:
In the documentation:

>  - `<robot_sim_type>`: The pluginlib name of a custom robot ros2_control system plugin to be used, defaults to `gazebo_ros2_control/GazeboSystem`

This is the plugin that you can rewrite based on your needs, you need to specify this in the plugin's parameters